src/components/studies/app-design/AppDesign.tsx

```diff
@@ -297,9 +300,30 @@ const AppDesign: React.FunctionComponent<
     if (!event.target.files) {
       return
     }
+    setIsSettingStudyLogo(true)
     const file = event.target.files[0]
     const previewForImage = getPreviewForImage(file)
-    setPreviewFile(previewForImage)
+    try {
+      const uploadResponse = (await StudyService.editStudyLogo(
```

Comment:
You don't need to cast this as Study. Your editStudyLogo method should return a promise of the type of study.